tests/phpunit/tests/user.php

```diff
@@ -604,6 +604,218 @@ public function test_count_many_users_posts() {
 		$this->assertSame( '1', $counts[ $user_id_b ] );
 	}
 
+	/**
+	 * Ensure the second and subsequent calls to count_many_users_posts() are cached.
+	 *
+	 * @ticket 63405
+	 */
+	public function test_count_many_users_posts_is_cached() {
+		$user_1 = self::$user_ids[0];
+		$user_2 = self::$user_ids[1];
+
+		// Create posts for both users.
+		self::factory()->post->create( array( 'post_author' => $user_1 ) );
+		self::factory()->post->create( array( 'post_author' => $user_2 ) );
+
+		// Warm the cache.
+		count_many_users_posts( array( $user_1, $user_2 ), 'post', false );
+
+		// Ensure cache is hit for second call.
+		$start_queries = get_num_queries();
+		count_many_users_posts( array( $user_1, $user_2 ), 'post', false );
+		$end_queries = get_num_queries();
+		$this->assertSame( $end_queries - $start_queries, 0, 'No database queries expected for second call to count_many_users_posts()' );
+	}
+
+	/**
+	 * Ensure equivalent arguments hit the same cache in count_many_users_posts().
+	 *
+	 * @ticket 63405
+	 *
+	 * @dataProvider data_count_many_users_posts_cached_for_equivalent_arguments
+	 *
+	 * @param array $first_args  First set of arguments to pass to count_many_users_posts().
+	 * @param array $second_args Second set of arguments to pass to count_many_users_posts().
+	 */
+	public function test_count_many_users_posts_cached_for_equivalent_arguments( $first_args, $second_args ) {
```

Comment:
Initially failing, see [test run for my first push](https://github.com/WordPress/wordpress-develop/actions/runs/18510850565/job/52751142938?pr=9106#step:14:499). Fixed by https://github.com/WordPress/wordpress-develop/pull/9106/commits/ca83326d7349263fe0d58032471360dde8414d8f